monai/application/handlers/stats_handler.py

```diff
@@ -0,0 +1,130 @@
+                                          must accept parameter "engine", use default printer if None.
+        iteration_print_logger (Callable): custimized callable printer for iteration level logging.
+                                         must accept parameter "engine", use default printer if None.
+
+    """
+    def __init__(self,
+                 epoch_print_logger=None,
+                 iteration_print_logger=None,
+                 extra_handler=None):
+        self.epoch_print_logger = epoch_print_logger
+        self.iteration_print_logger = iteration_print_logger
+        self.logger_id = 'stat_logger'
+        self.logger = logging.getLogger(logger_id)
+        self.logger.setLevel(logging.INFO)
+        stream_handler = logging.StreamHandler()
+        stream_handler.setLevel(logging.INFO)
+        self.logger.addHandler(stream_handler)
+        if extra_handler is not None:
+            self.logger.addHandler(extra_handler)
+
+    def attach(self, engine: Engine):
+        """Register a set of Ignite Event-Handlers to a specified Ignite engine.
+
+        Args:
+            engine (ignite.engine): Ignite Engine, it can be a trainer, validator or evaluator.
+
+        """
+        engine.add_event_handler(Events.EPOCH_COMPLETED, self.epoch_completed)
+        engine.add_event_handler(Events.ITERATION_COMPLETED, self.iteration_started)
+        engine.add_event_handler(Events.EXCEPTION_RAISED, self.exception_raised)
+
+    def epoch_completed(self, engine: Engine):
+        """handler for train or validation/evaluation epoch completed Event.
+        Print epoch level log, default values are from ignite state.metrics dict.
+
+        Args:
+            engine (ignite.engine): Ignite Engine, it can be a trainer, validator or evaluator.
+
+        """
+        if self.epoch_print_logger is not None:
+            self.epoch_print_logger(engine)
+        else:
+            self._default_epoch_print(engine)
+
+    def iteration_completed(self, engine: Engine):
+        """handler for train or validation/evaluation iteration completed Event.
+        Print iteration level log, default values are from ignite state.logs dict.
+
+        Args:
+            engine (ignite.engine): Ignite Engine, it can be a trainer, validator or evaluator.
+
+        """
+        if self.iteration_print_logger is not None:
+            self.iteration_print_logger(engine)
+        else:
+            self._default_ieration_print(engine)
+
+    def exception_raised(self, engine: Engine, e):
+        """handler for train or validation/evaluation exception raised Event.
+        Print the exception information and traceback.
+
+        Args:
+            engine (ignite.engine): Ignite Engine, it can be a trainer, validator or evaluator.
+            e (Exception): the exception catched in Ignite during engine.run().
+
+        """
+        self.logger.exception('Exception: {}'.format(e))
+        traceback.print_exc()
```

Comment:
This prints to stderr, should we instead print the trace to the logger or not at all?